Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1123 @@
+
+        $FrontDoorResources | ForEach-Object {
+                $frontDoorResource = $_
+                $resourceGroupName = $_.ResourceGroupName
+                $resourceName = $_.Name
+                try
+                {
+                    if($frontDoorResource.MinimumTlsVersion -eq $null)
+                    {  
+                    Write-Host "Fetching Front Door Endpoint configuration: Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+                    Write-Host $([Constants]::SingleDashLine)
+                    $FrontDoorCustomDomains = @()
+
+                    $FrontDoorCustomDomains = Get-AzFrontDoorCdnCustomDomain -ProfileName $resourceName -ResourceGroupName $resourceGroupName
+                
+                    if($FrontDoorCustomDomains)
+                    {
+                        Write-Host "Front Door Configurations successfully fetched." -ForegroundColor $([Constants]::MessageType.Update)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+
+                    if($null -ne $FrontDoorCustomDomains)
+                    {
+                        foreach ($item in $FrontDoorCustomDomains) 
+                        {
+                            $frontDoorEp = [FrontendEndPointsClass]:: new()
+                            $apiError = $false
+                            $domainName = $item.Name
+                            $domainResourceGroupName = $item.ResourceGroupName
+                            $TypeOfCertificate = $null
+                            $MinTLSVersion = $null
+                    
+                            if($null -ne $AccessMgtToken)
+                            {
+                                $header = "Bearer " + $AccessMgtToken
+                                $headers = @{"Authorization"=$header;"Content-Type"="application/json"; "x-ms-version" ="2013-08-01"}
+                                $uri = [string]:: Format("{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.Cdn/profiles/{3}/customDomains/{4}?api-version=2021-06-01",$ResourceAppIdURI,$SubscriptionId,$domainResourceGroupName,$resourceName,$domainName)
+                                $customDomainResponse = Invoke-WebRequest -Method GET -Uri $uri -Headers $headers -UseBasicParsing
+                    
+                                if($customDomainResponse.StatusCode -ge 200 -and $customDomainResponse.StatusCode -le 399)
+                                {
+                                    if($null -ne $customDomainResponse.Content)
+                                    {
+                                        $custom = $customDomainResponse.Content | ConvertFrom-Json  
+                                        if($null -ne $custom)
+                                        {
+                                            $TypeOfCertificate = $custom.properties.tlsSettings.certificateType
```

Comment:
please set these two variables to null in starting of loop.

Reply:
It is already null when they are defined.
$TypeOfCertificate = $null
$MinTLSVersion = $null